### DISCUSSION TITLE
Default field value still required for a relation field?

### DISCUSSION THREAD

**Discussion by TomasMoskala** [prisma/prisma#2716] [Q&A] [answered]
Hi, 
why does a relation field require a value although it has a default one? 
It even works without connect / create in the playground.
In resolvers it can can be removed from args but not from data - there we have to hardcode the default value again?

**Comment by ryands17** [comment]
Hey @TomasMoskala :wave: 
Could you share the schema so that I could check the relation that you are mentioning about?

> In resolvers it can can be removed from args but not from data - there we have to hardcode the default value again?

Also what do you mean by cannot be removed from data?

**Reply by TomasMoskala** [reply]
Make authorID not nullable with a default
```prisma
model Post {
  id        Int     @default(autoincrement()) @id
  authorId  Int     @default(1)
  content   String?
  published Boolean @default(false)
  title     String
  author    User   @relation(fields: [authorId], references: [id])
}
```
You can remove it from args but cannot  remove it from ctx.prisma.post.create

```ts
resolve: (_, { title, content, _authorEmail_ }, ctx) => {
        return ctx.prisma.post.create({
          data: {
            title,
            content,
            published: false,
            **author: {
              connect: { email: authorEmail },**
            },
          },
        })
```

you must go
author: { connect: { id:1 },
although the field already has a default value

The syntax check still cries but it works
![image](https://user-images.githubusercontent.com/35471478/84366220-d009b380-abd2-11ea-9f7d-087b1e12d36c.png)

![image](https://user-images.githubusercontent.com/35471478/84363680-61772680-abcf-11ea-9cef-2b1b1f72069e.png)

**Reply by ryands17** [reply, marked as answer]
It works if you keep the relation optional in the following manner:

```prisma
model Post {
  id        Int     @default(autoincrement()) @id
  authorId  Int?     @default(1)
  content   String?
  published Boolean @default(false)
  title     String
  author    User?   @relation(fields: [authorId], references: [id])
}
```

This will make it optional and add the default when required.

Just a side note: This would cause an error if the default supplied foreign key doesn't exist.

**Reply by ryands17** [reply]
@TomasMoskala does the above solution work for you?

**Reply by TomasMoskala** [reply]
That entails using nullable required fields, I don't like it much...

**Comment by ludralph** [comment]
Hi @TomasMoskala,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
